main/bluetooth/hidp/ps.c

```diff
@@ -45,6 +47,76 @@ static void bt_hid_cmd_ps5_rumble_init(struct bt_dev *device) {
     bt_hid_cmd_ps5_set_conf(device, (void *)&ps5_set_conf);
 }
 
+static void bt_hid_cmd_ps5_trigger_init(struct bt_dev *device) {
+    int32_t perc_threshold_l = -1;
+    int32_t perc_threshold_r = -1;
+    int32_t dev = bt_host_get_dev_from_out_idx(device->ids.out_idx, &device);
+    uint32_t map_cnt_l = 0;
+    uint32_t map_cnt_r = 0;
+
+    /* Go through the list of mappings, looking for PAD_RM and PAD_LM */
+    for (uint32_t i = 0; i < config.in_cfg->map_size; i++) {
+        /* Threshold doesn't make sense when the destination is axis, skip */
+        if (btn_is_axis(config.in_cfg[dev].map_cfg[i].dst_id, config.in_cfg[dev].map_cfg[i].dst_btn)) {
+            continue;
+        }
+        /* Destination is button, set the "click" at its assigned threshold */
+        if (config.in_cfg[dev].map_cfg[i].src_btn == PAD_RM) {
+            if (config.in_cfg[dev].map_cfg[i].perc_threshold > perc_threshold_r) {
```

Comment:
Since we can have multiple mapping using the same src, I decided to use the mapping with the highest threshold.

But thinking about it again now, maybe it's should be 2nd lowest? But for simplicity I think I leave it like that.

Reply:
I was thinking about it. I think highest threshold makes sense. Before, my code had it use the last mapping.